google_compute_engine/metadata_scripts/tests/script_retriever_test.py

```diff
@@ -44,6 +44,21 @@ def testDownloadGsNoExec(self, mock_call):
 
   @mock.patch('google_compute_engine.metadata_scripts.script_retriever.subprocess.check_call')
   @mock.patch('google_compute_engine.metadata_scripts.script_retriever.tempfile.NamedTemporaryFile')
+  def testDownloadGsUrlFallback(self, mock_tempfile, mock_call):
```

Comment:
Any ideas on how to prevent `gsutil` from running during the test and causing it to fetch other endpoint?
I think we can basically check if disabling `gsutil` still returns the original URL on a few cases